### PR TITLE
Sync changes from NeurIPS experiments

### PR DESCRIPTION
**Pull request by davmre**
This PR includes all changes made internally to run experiments for NeurIPS 2019, and some miscellaneous cleanups. Squashing all these changes into one PR might not be the best Github hygiene (but it's easy) -- happy to break them up if you prefer to review them separately. :-)

This version of the code depends on some recent functionality in TFP (the SimpleStepSizeAdaptation kernel), which will be part of the upcoming 0.7.0 release; (and is currently in tfp-nightly). I've added a note to the README mentioning that the code requires TFP 0.7.

### CODE REVIEW

**Comment by davmre** [comment]
Hi Maria, I'm kind of rusty on git best practices, but figured rather than just blindly push to master (like I accidentally did with the formatting fixes when I first imported this code back in), I might as well go through proper channels this time. These changes should be mergeable as-is, but let me know if anything looks suspect to you.

File: analyze.py [hunk 1]
```diff
@@ -42,25 +72,31 @@ def get_elbos(model_name):
...
+    file_path = os.path.join(FLAGS.results_dir, model_name, m + ".json")
 
     with tf.gfile.Open(file_path, "r") as f:
       results = json.load(f)
       reparam = results["learned_reparam"]
 
+    parameterisation[m] = dict([(key,
+                                 (np.array(reparam[key])).astype(np.float32))
+                                for key in reparam.keys()])
     discrete_parameterisation[m] = dict(
```
Comment (mgorinova) [inline analyze.py:87]: Calculating `discrete_parameterisation` becomes redundant here.

File: analyze.py [hunk 2]
```diff
@@ -42,25 +72,31 @@ def get_elbos(model_name):
 def get_reparam(model_name):
 
   discrete_parameterisation = {}
```
Comment (mgorinova) [inline analyze.py:74]: `discrete_parameterisation` is redundant here.

**Review by mgorinova — APPROVED** [review]
Thanks so much for all the changes, Dave! I had a look and all looks good. Two minor comments about redundant lines. Approving this and merging; will fix the lines separately.

**Pull request merged**